Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -326,8 +326,31 @@ WebProcessPool::WebProcessPool(API::ProcessPoolConfiguration& configuration)
 
 }
 
+void WebProcessPool::keepGPUProcessAliveTemporarily()
+{
+#if ENABLE(GPU_PROCESS)
+    if (!m_gpuProcess)
+        return;
+
```

Comment:
Should we `ASSERT` that we are on the main thread for this logic?

Reply:
Sure, I'll add an assertion.